### PR TITLE
PSP-4462 Update financial code

### PR DESCRIPTION
**Pull request by asanchezr**
image

### CODE REVIEW

File: source/backend/api/Services/FinancialCodeService.cs [hunk 1]
```diff
@@ -133,7 +152,93 @@ public FinancialCodeModel Add(FinancialCodeTypes type, FinancialCodeModel model)
...
+        }
+
+        public FinancialCodeModel Update(FinancialCodeTypes type, FinancialCodeModel model)
+        {
+            model.ThrowIfNull(nameof(model));
+
+            _logger.LogInformation("Updating financial code with type {type} and model {model}", type, model);
+            User.ThrowIfNotAuthorized(Permissions.SystemAdmin);
+
+            ValidateVersion(type, model.Id, model.RowVersion);
```
Comment (devinleighsmith) [inline source/backend/api/Services/FinancialCodeService.cs:166]: What is the benefit of checking this manually? The db will throw a dbconcurrency exception if the rowversion is not updated.

**Comment by devinleighsmith** [comment]
Looks like this is missing tests on the backend. Either need that for this story or need tech debt logged to cover missing coverage.

**Comment by asanchezr** [comment]
> Looks like this is missing tests on the backend. Either need that for this story or need tech debt logged to cover missing coverage.

As mentioned in standup I'm working in them but the rest of the code is reviewable

File: source/backend/api/Services/FinancialCodeService.cs (hunk 1 above)
Comment (asanchezr) [reply source/backend/api/Services/FinancialCodeService.cs:166]: Well, this pattern is pervasive across other services. If we want to abandon it then we should do it across all of them. Topic for Friday chat I think

Comment (asanchezr) [reply source/backend/api/Services/FinancialCodeService.cs:166]: This is a snap of another service doing the check.
image

Comment (FuriousLlama) [reply source/backend/api/Services/FinancialCodeService.cs:166]: I think the original intent of that method was to be used by child entities to make sure the parent was in sync

**Review by devinleighsmith — APPROVED** [review]

**Pull request merged**